_posts/2018-04-11-sign-to-contract.md

```diff
@@ -0,0 +1,400 @@
+The former is the signature corresponding to the pubkey `x*G` for the message `m`,
+while the latter and `c` are the ingredients to prove the commitment to `Q.x`. 
+Now `uTX` is filled with `SIG(q,z)` which contains `q`, which in turn is (almost always) `Q.x`.
+As before, the resulting `TX` then is committed to the blockchain.
+
+Carl then looks at `TX`, in which he spots `Q.x`, with `TX = TXp||Q.x||TXa`. 
+By retrieving the info to link `TX` to a block header we can complete the timestamp for the contract `c`:
+```
+File sha256: c
+Timestamp:
+prepend R == R||c
+secp256k1commitment == Q.x
+prepend TXp == TXp||Q.x
+append TXa == TXp||Q.x||TXa
+# transaction id ...
+sha256
+...
+sha256
+verify BitcoinBlockHeaderAttestation(...)
+# Bitcoin merkle root ...
+``` 
+**Resuming**, while doing a purely financial transaction to Diana, **Carl timestamped** a contract, 
+without adding any bytes, so **with zero marginal cost**.
+Diana could be unaware that Carl timestamped, 
+from her prospective the coins she just received are indistinguishable from coins signed in the standard way. 
+Such coins are locked in the pubkey (or address) that Diana told Carl, 
+hence Diana's new coins are as secured as normal coins. 
+
+But what about Carl?
+Aside from timestamping, can this procedure compromise his security?
+
+The delicate phase is between the broadcast and the confirmation, 
+if, in that period, Carl secret key `x` gets exposed, an eavesdropper, Eve, will redirect Carl's coin to her. 
+However, once the transaction to Diana is confirmed, Eve cannot do much (unless Carl does address reuse), 
+since the coins locked by `x` are already spent.
+The private key can also be revealed in a indirect way, 
+in fact in a signature revealing the nonce `k` 
+(or `e` to be consistent with the above formula) 
+is equivalent to expose `x`.
+Still *sign-to-contract* does not increase the risks Carl faces with a standard signature.
+Indeed when showing his OTS proof he just claims that what can be found  in the signature (namely `Q.x`) was obtained from the elliptic curve point (`Q=e*G`) which is the sum of two other specific points (`R` and `h(R||c)*G`),
+but actually this does not give any additional clue to solve the discrete logarithm that secures the signature.
+
+Another way to expose `x` is signing two times with two different nonces
+and then spreading both signatures;
+this may happen if Carl uses *sign-to-contract* with 
+`(x,m,c1)` and `(x,m,c2)` and then broadcasts both transactions.
+He should take care not to do so.
+
+In addition, if Carl looses the contract along with its hash `c`, 
+then he looses the ability of proving the timestamp,
+but he does not compromise any coin: 
+his coins have been already spent and Diana has hers in the desired destination, 
+which is independent from the contract.
+For this reason *sign-to-contract* should be preferred when timestamping.
+
+## Remarks
+- The name contract is merely due to historical reasons.
```

Comment:
You could avoid this terminology issue by - like I mentioned above - introducing the whole concept as a secp256k1 *commitment*, and if you want, noting how historically it came out of a desire to *commit* to contracts when paying someone.

Reply:
Many thanks for all the comments, I should have fixed all of those here 8d415e3bf5fbd2a9670e35cc4d3532feabc00d71.